streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/HighAvailabilityTaskAssignor.java

```diff
@@ -43,21 +43,27 @@
 
 public class HighAvailabilityTaskAssignor implements TaskAssignor {
     private static final Logger log = LoggerFactory.getLogger(HighAvailabilityTaskAssignor.class);
+    private static final int DEFAULT_STATEFUL_TRAFFIC_COST = 10;
+    private static final int DEFAULT_STATEFUL_NON_OVERLAP_COST = 1;
+    private static final int DEFAULT_STATELESS_TRAFFIC_COST = 1;
+    private static final int DEFAULT_STATELESS_NON_OVERLAP_COST = 1;
```

Comment:
We have different costs for stataful vs stateless here, but the configs added only allow to pass in one value each for traffic and non-overlap. How does this fix together?

From the code below it seem, users only configure cost for stateful anyway, but stateless cost is alway `1` (for this case, the variables should not be called `DEFAULT` because they cannot be changed).

Also, if we have default value, should they not be set in `StreamsConfig`?

Reply:
Yeah. The only stateful cost is configurable. The default is not set in `StreamsConfig` because it will be different for `HAAssignor` and `StickyAssignor`. So the default in `StreamsConfig` is null which means use assignor's default.